packages.config

```diff
@@ -3,4 +3,5 @@
   <package id="Antlr4.Runtime" version="4.5.3" targetFramework="net46" />
   <package id="log4net" version="2.0.5" targetFramework="net45" />
   <package id="MicroBuild.Core" version="0.2.0" targetFramework="net46" developmentDependency="true" />
+  <!-- <package id="Microsoft.VSSDK.BuildTools" version="15.8.3247" developmentDependency="true" /> -->

```

Comment:
Why did this have to be disabled?
Why is this commented out and not just removed?
Does removing VSSDK dependency here means VSSDK must be installed manually through VS installer?

Reply:
This is a development artifact. I was having trouble in AppVeyor and assumed that it was missing the VSSDK build tools when in fact was a matter of stating `<CopyVsixManifestToOutput>false</CopyVsixManifestToOutput>` in BoostTestPackage.csproj.

I left it commented out since it might have value to revise this aspect but given the effort in switching to using `<PackageReference>` I was not convinced this was the right call. I am also not 100% familiar with the VS installation procedure and I am not sure if the build tools are also packaged with every VS version. Regardless, switching to using the NuGet package would also require amending the projects to reference the property sheets from the respective locations.

It might be better to remove than leave as is since I understand that it may get confusing. Let me know how best to proceed.

Reply:
AFAIR one can consume VSSDK for extension development either implicitly (i.e. using named references to assemblies, which are expected to be resolved automatically, but require explicitly installing the "Visual Studio extension development" workload) or explicitly (i.e. through NuGet, be it packages.config or package references). When we were last time talking to teams (co-)owning the SDK, the latter was the preferred approach. Having said that, making the switch is some more work as you have observed, so while contributions in that area are welcome, they would probably make sense to form a separate PR.

For historical reference, I should also note that we attempted in the past to switch to package references format of NuGet dependencies, but ran into some problems with our internal infrastructure we use for official builds (it has some additional quirks around signing and localizing the binaries, that we do not usually need worry about in the regular development environment, but matter when releasing into official channels) and ended up rolling the changes back (#15) to expedite a particular release. I am sure these issues were not insurmountable, but we would need to tackle them if we re-attempt the switch. Or maybe the infrastructure had improved in the meantime so that the issues are no longer there?

Orthogonal to all that, we generally hold the rule that commented out code should not be committed, unless it as accompanied with a descriptive comment indicating why the code is commented out, and preferably an issue number tracking resolution of any reasons for the code not be active. So I think in this case you might want to just undo changes to this file.